include/picongpu/simulation/control/Simulation.hpp

```diff
@@ -486,27 +488,7 @@ namespace picongpu
                     initialiserController->restart((uint32_t) this->restartStep, this->restartDirectory);
                     step = this->restartStep;
 
-                    /** restore background fields in GUARD
-                     *
-                     * loads the outer GUARDS of the global domain for absorbing/open boundary condtions
-                     *
-                     * @todo as soon as we add GUARD fields to the checkpoint data, e.g. for PML boundary
-                     *       conditions, this section needs to be removed
-                     */
-                    cellwiseOperation::CellwiseOperation<GUARD> guardBGField(*cellDescription);
-                    namespace nvfct = pmacc::nvidia::functors;
-                    guardBGField(
-                        fieldE,
-                        nvfct::Add(),
-                        FieldBackgroundE(fieldE->getUnit()),
-                        step,
-                        FieldBackgroundE::InfluenceParticlePusher);
-                    guardBGField(
-                        fieldB,
-                        nvfct::Add(),
-                        FieldBackgroundB(fieldB->getUnit()),
-                        step,
-                        FieldBackgroundB::InfluenceParticlePusher);
+                    fieldBackground.fillSimulation(step);
```

Comment:
I tried to verify that all field including guard/core/border is always in the current time step.
I would say that we break the restart of a simulation with field background enabled.

The plugin output including checkpoint will always contain the fields + background. After a restart we need too remove the field background from the checkpointed data.

@sbastrakov could we discuss it in a VC, the problem that we have the background field included in the checkpoints too will introduce numerical artefacts we tried to avoid with the new method. We should brain storm how we taggle this issue.

Reply:
As discussed in a VC, how checkpoints interact with `Simulation` and data in general may require some reworking to make it more simple. Currently checkpoints incorporated some changes and workarounds, which results in some not very natural things, like field background being added in the moving window check, also with a workaround for when we have just loaded a checkpoint. We discuss how to implement this PR within the existing workflow of checkpointing. And that workflow may be improved later in a separate PR.

Reply:
The operation indeed did nothing, as we thought during the VC as it's always overridden by the following field exchange. I have removed it, it also allowed to simplify some core, as it no longer needs to parametrize on area.

With respect to operating right after loading from a checkpoint, this is also fixed now.